cmd/adsysd/client/policy.go

```diff
@@ -383,20 +410,73 @@ func (a *App) update(isComputer, updateAll bool, target, krb5cc string) error {
 	return nil
 }
 
-func (a App) completeWithConnectedUsers() ([]string, cobra.ShellCompDirective) {
+func (a *App) purge(isComputer, purgeAll bool, target string) error {
+	// incompatible options
+	if purgeAll && target != "" {
+		return errors.New(i18n.G("machine or user arguments cannot be used with update all"))
+	}
+	if isComputer && target != "" {
+		return errors.New(i18n.G("user arguments cannot be used with machine update"))
+	}
+
+	client, err := adsysservice.NewClient(a.config.Socket, a.getTimeout())
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	// get target for computer
+	if isComputer && target == "" {
+		hostname, err := os.Hostname()
+		if err != nil {
+			return err
+		}
+		// for malconfigured machines where /proc/sys/kernel/hostname returns the fqdn and not only the machine name, strip it
+		target, _, _ = strings.Cut(hostname, ".")
+	}
+
+	// Purge current user
+	if target == "" && !purgeAll {
+		u, err := user.Current()
+		if err != nil {
+			return fmt.Errorf("failed to retrieve current user: %w", err)
+		}
+		target = u.Username
+	}
+
+	stream, err := client.UpdatePolicy(a.ctx, &adsys.UpdatePolicyRequest{
+		IsComputer: isComputer,
+		All:        purgeAll,
+		Target:     target,
+		Purge:      true,
+	})
+	if err != nil {
+		return err
+	}
+
+	if _, err := stream.Recv(); err != nil && !errors.Is(err, io.EOF) {
+		return err
+	}
+
+	return nil
+}
+
+// connectedUsers returns the list of connected users according to their cached policy information.
+// If active is true, the list of users is retrieved from the cached Kerberos ticket information.
+func (a App) connectedUsers(active bool) []string {
```

Comment:
So, maybe connectedUsers is wrongly named now? I think connected/active convey the same meaning, wdyt?

Reply:
sounds good to me